rotkehlchen/chain/ethereum/modules/yearn/decoder.py

```diff
@@ -0,0 +1,128 @@
+from rotkehlchen.chain.ethereum.modules.yearn.constants import CPT_YEARN_V1, CPT_YEARN_V2
+from rotkehlchen.chain.evm.constants import ZERO_ADDRESS
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import ActionItem
+from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+from rotkehlchen.constants.resolver import ethaddress_to_identifier
+from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.types import (
+    YEARN_VAULTS_V1_PROTOCOL,
+    YEARN_VAULTS_V2_PROTOCOL,
+    ChainID,
+    EvmTransaction,
+)
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.manager import EthereumInquirer
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.types import ChecksumEvmAddress
+    from rotkehlchen.user_messages import MessagesAggregator
+
+YEARN_DEPOSIT_4_BYTES = b'\xb6\xb5_%'
+YEARN_WITHDRAW_4_BYTES = b'.\x1a}M'
+
+
+def _get_vault_token_name(vault_address: 'ChecksumEvmAddress') -> str:
+    try:
+        vault_token = EvmToken(ethaddress_to_identifier(vault_address))
+        vault_token_name = vault_token.name
+    except (UnknownAsset, WrongAssetType):
+        vault_token_name = str(vault_address)
+
+    return vault_token_name
+
+
+class YearnDecoder(DecoderInterface):
+    def __init__(
+            self,
+            ethereum_inquirer: 'EthereumInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(ethereum_inquirer, base_tools, msg_aggregator)
+        self.base_tools = base_tools
+        vaults_v1 = GlobalDBHandler().get_evm_tokens(chain_id=ChainID.ETHEREUM, protocol=YEARN_VAULTS_V1_PROTOCOL)  # noqa: E501
+        vaults_v2 = GlobalDBHandler().get_evm_tokens(chain_id=ChainID.ETHEREUM, protocol=YEARN_VAULTS_V2_PROTOCOL)  # noqa: E501
+        self.vaults_v1 = {vault.evm_address for vault in vaults_v1}
+        self.vaults_v2 = {vault.evm_address for vault in vaults_v2}
+
+    def _maybe_enrich_yearn_transfers(
+            self,
+            token: 'EvmToken',  # pylint: disable=unused-argument
+            tx_log: EvmTxReceiptLog,  # pylint: disable=unused-argument
+            transaction: EvmTransaction,
+            event: HistoryBaseEntry,
+            action_items: list[ActionItem],  # pylint: disable=unused-argument
+    ) -> bool:
+        counterparty = CPT_YEARN_V2
+        if transaction.to_address in self.vaults_v1:
+            counterparty = CPT_YEARN_V1
+        elif transaction.to_address not in self.vaults_v2:
+            return False
+
+        is_deposit = False
+        if transaction.input_data.startswith(YEARN_DEPOSIT_4_BYTES):
+            is_deposit = True
+        elif not transaction.input_data.startswith(YEARN_WITHDRAW_4_BYTES):
+            # in this case is a method that we don't handle
```

Comment:
Why?

```suggestion
            # in this case it is a method that we don't handle
```

Reply:
In this case as the logic is written it means that is not a `deposit` neither ` withdraw` since we are checking the function called in the transaction. We are checking before this that we are calling a yearn contract and with this logic we ensure that only deposit/withdrawals are processed. The contract has other methods that I'm not attempting to decode in this PR